components/core/tests/test-NetworkReader.cpp

```diff
@@ -193,6 +193,9 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
 }
 
 TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
+    // Since the http request sometimes crash due to arbitrary reasons, so we just try it
+    // multiple times to avoid such cases
+    constexpr int cNumMaxTrials{10};
```

Comment:
```suggestion
    constexpr size_t cNumMaxTrials{10};
```
Avoid using `int` for counter like variables